syntax = "proto3";

package chat;

service ChatService {
    rpc JoinChat (JoinRequest) returns (stream ChatMessage);
    rpc SendMessage (ChatMessage) returns (Empty);
}

message JoinRequest {
    string username = 1;
}

message ChatMessage {
    string username = 1;
    string text = 2;
}

message Empty {}
